src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -150,6 +153,33 @@ static int client_receive_message_udp(
                 void *userdata);
 static void client_stop(sd_dhcp_client *client, int error);
 
+int sd_dhcp_client_id_to_string(char **ret, const void *data, size_t len) {
+        const sd_dhcp_client_id *client_id = data;
+
+        switch (client_id->type) {
+        case 0:
+                if (utf8_is_printable((char *)client_id->gen.data, len)) {
+                        return asprintf(ret, "%.*s", (int) len, client_id->gen.data);
+                }
+                return asprintf(ret, "DATA");
+        case 1:
+                return asprintf(ret, "%x:%x:%x:%x:%x:%x",
+                                client_id->eth.haddr[0],
+                                client_id->eth.haddr[1],
+                                client_id->eth.haddr[2],
+                                client_id->eth.haddr[3],
+                                client_id->eth.haddr[4],
+                                client_id->eth.haddr[5]);
+        case 2 ... 254:
+                return asprintf(ret, "ARP/LL");
+        case 255:
+                return asprintf(ret, "IAID:0x%x/DUID", be32toh(client_id->ns.iaid));
+        }
+
+        return -1;
```

Comment:
`type` has range 0..255. This code is unreachable.

Reply:
also always return proper negative errno-style errors on failure, i.e. never make up "-1" as an error.

Reply:
ok